signer/apk2/apk2.go

```diff
@@ -0,0 +1,148 @@
+package apk2
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"os"
+	"os/exec"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"go.mozilla.org/autograph/signer"
+)
+
+const (
+	// Type of this signer is "apk2" represents a signer that
+	// shells out to apksigner to sign artifacts
+	Type = "apk2"
```

Comment:
@jvehent what do you think about adding a version field to signers in the future?

That way we could avoid adding and removing whole signers for e.g. rsapss to genericrsa, apk to apk2, and contentsignature to contentsignaturepki. Also we could use that field to pick the underlying signing binary for the gpg2 and apk2 signers e.g. for apksigner updates in the future.

Reply:
I need to think a little bit about how that would work. Wouldn't it make the logic inside the signer a lot more complicated though? And the tests as well?

Reply:
> Wouldn't it make the logic inside the signer a lot more complicated though? And the tests as well?

More paths to cover, but less code :woman_shrugging:

I can split out another issue to discuss.